packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -17,6 +17,424 @@ const GraphQLJSON = require('graphql-type-json');
+      };
+      return resolver(object);
+    };
+  },
+
+  /**
+   * Create fields resolvers
+   * 
+   * @return {Object}
+   */
+  createFieldsResolver: function(fields, resolver, typeCheck) {
+    return _.reduce(fields, (acc, field, key) => {
+      // Check if the field is of the correct type
+      if (typeCheck(field)) {
+        return _.set(acc, key, (obj, options, context) => {
+          return resolver(obj, options, context, this.fieldResolver(field, key), key, obj, field);
+        });
+      }
+      return acc;
+    }, {});
+  },
+
+  /**
+   * Build the mongoose aggregator by applying the filters
+   */
+  getModelAggregator: function (model, filters = {}) {
+    const aggregation = model.aggregate();
+    if (!_.isEmpty(filters.where)) {
+      aggregation.match(filters.where);
+    }
+    if (filters.limit) {
+      aggregation.limit(filters.limit);
+    }
+    return aggregation;
+  },
+
+  /**
+   * Create the resolvers for each aggregation field
+   * 
+   * @return {Object}
+   * 
+   * @example
+   * 
+   * const model = // Strapi model
+   * 
+   * const fields = {
+   *   username: String,
+   *   age: Int,
+   * }
+   * 
+   * const typeCheck = (type) => type === 'Int' || type === 'Float',
+   * 
+   * const fieldsResoler = createAggregationFieldsResolver(model, fields, 'sum', typeCheck);
+   * 
+   * // => {
+   *   age: function ageResolver() { .... }
+   * }
+   */
+  createAggregationFieldsResolver: function (model, fields, operation, typeCheck) {
+    return this.createFieldsResolver(fields, async (obj, options, context, fieldResolver, fieldKey) => { // eslint-disable-line no-unused-vars
+      const result = await this.getModelAggregator(model, obj).group({
+        _id: null,
+        [fieldKey]: { [`$${operation}`]: `$${fieldKey}` }
+      });
+      return _.get(result, `0.${fieldKey}`);
+    }, typeCheck);
+  },
+
+  /**
+   * Correctly format the data returned by the group by
+   */
+  preProcessGroupByData: function ({ result, fieldKey, filters, modelName }) {
+    const _result = _.toArray(result);
+    return _.map(_result, (value) => {
+      const params = Object.assign(
+        {},
+        this.convertToParams(_.omit(filters, 'where')),
+        filters.where,
+        {
+          [fieldKey]: value._id,
+        }
+      );
+
+      return {
+        key: value._id,
+        connection: strapi.utils.models.convertParams(modelName, params),
+      };
+    });
+  },
+
+  /**
+   * Create the resolvers for each group by field
+   * 
+   * @return {Object}
+   * 
+   * @example
+   * 
+   * const model = // Strapi model
+   * const fields = {
+   *   username: [UserConnectionUsername],
+   *   email: [UserConnectionEmail],
+   * }
+   * const fieldsResoler = createGroupByFieldsResolver(model, fields);
+   * 
+   * // => {
+   *   username: function usernameResolver() { .... }
+   *   email: function emailResolver() { .... }
+   * }
+   */
+  createGroupByFieldsResolver: function (model, fields, name)  {
```

Comment:
I need to make more tests with a clean setup but it's seems to not working.

<img width="1440" alt="capture d ecran 2018-06-19 a 16 44 39" src="https://user-images.githubusercontent.com/2202989/41604705-2b563218-73e0-11e8-8c7e-770ee0a16951.png">